File: small-programs/IsLeapYear.java
/* *****************************************************************************
 *  Name:              Ada Lovelace
 *  Coursera User ID:  123456
 *  Last modified:     October 16, 1842
 * A leap year is (1) divisible by 4 but not 100 or (2) divisible by 400
 **************************************************************************** */

public class IsLeapYear {
    public static void main(String[] args) {
        int year = Integer.parseInt(args[0]);
        boolean isLeap;
        isLeap = (year % 4 == 0) && (year % 100 != 0);
        isLeap = isLeap || (year % 400 == 0);
        System.out.println(isLeap);
    }
}

File: small-programs/is_leap_year.py
"""Return True/False if a given year is a leap year.
A leap year is either divisible by 4 but not by 100 or divisible by 400."""

import sys

year = int(sys.argv[1])

is_leap = (year % 4 == 0) and (year % 100 != 0)
is_leap = is_leap or (year % 400 == 0)

print(is_leap)